components/global/global_header.tsx

```diff
@@ -1,82 +1,131 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import ThemeProvider, {lightTheme} from '@mattermost/compass-components/utilities/theme';
-import React from 'react';
+import React, {useEffect} from 'react';
 import {useSelector} from 'react-redux';
 
 import styled from 'styled-components';
 
-import {getGlobalHeaderEnabled} from 'selectors/global_header';
 import StatusDropdown from 'components/status_dropdown';
-
+import {getGlobalHeaderEnabled} from 'selectors/global_header';
 import Pluggable from 'plugins/pluggable';
 
+import GlobalSearchNav from './global_search_nav';
 import ProductSwitcher from './product_switcher';
+import HistoryButtons from './history_buttons';
+import UserGuideDropdown from './user_guide_dropdown';
+import AtMentionsButton from './at_mentions_button/';
+import SavedPostsButton from './saved_posts_button/';
+import SettingsButton from './settings_button//';
+
 import {useCurrentProductId, useProducts} from './hooks';
 
-const HeaderContainer = styled.div`
+import './global_header.scss';
+
+const GlobalHeaderContainer = styled.header`
     position: relative;
     display: flex;
-    flex-direction: row;
+    flex-shrink: 0;
     align-items: center;
+    justify-content: space-between;
     height: 40px;
-    background: var(--sidebar-teambar-bg);
-    color: var(--sidebar-text);
+    background: var(--global-header-background);
+    border-bottom: solid 1px rgba(var(--center-channel-color-rgb), 0.08);
+    color: rgba(var(--global-header-text-rgb), 0.64);
+    padding: 0 16px 0 12px;
+
+    > * + * {
+        margin-left: 12px;
+    }
+`;
+
+const LeftControls = styled.div`
+    display: flex;
+    align-items: center;
+    height: 40px;
+    flex-shrink: 0;
+
+    > * + * {
+        margin-left: 12px;
+    }
 `;
 
-const AppSpectificContent = styled.div`
+const CenterControls = styled.div`
+    display: flex;
+    align-items: center;
+    height: 40px;
+    justify-content: center;
     flex-grow: 1;
+
+    > * + * {
+        margin-left: 8px;
+    }
 `;
 
-const ProfileWrapper = styled.div`
-    margin-right: 20px;
+const RightControls = styled.div`
+    display: flex;
+    align-items: center;
+    height: 40px;
+    flex-shrink: 0;
+
+    > * + * {
+        margin-left: 8px;
+    }
 `;
 
 const GlobalHeader = (): JSX.Element | null => {
     const enabled = useSelector(getGlobalHeaderEnabled);
     const products = useProducts();
     const currentProductID = useCurrentProductId(products);
 
+    useEffect(() => {
```

Comment:
This is fragile. Maybe we could use styled components for this?

Reply:
I guess it was implemented that way due to the fact that the feature flag will be removed soon. I can bring it up for discussion in out team meeting today.

Reply:
@crspeller, this is needed to update styles throughout the rest of the app that need to change only when the Global Header is enabled. It is the equivalent of a feature flag for JS but makes it easier to group styles together with the Global Header vs. trying to keep track of and update all the differences in the various SASS files spread throughout the code base and support the GH being on or off. It is not possible to convert everything outside the Global Header that need CSS changes when the GH is on given the available time.

This absolutely should be replaced with applying styling changes directly when the Global Header is no longer a feature.